tjpcov_new/covariance_io.py

```diff
@@ -0,0 +1,147 @@
+        It does not compute anything.
+
+        Parameters
+        ----------
+        config (dict or str):
+        """
+        self.config = self._read_config(config)
+        self.sacc_file = None
+
+        # Output directory where to save all the time consuming calculations
+        self.outdir = self.config['tjpcov'].get('outdir', './')
+        os.makedirs(self.outdir, exist_ok=True)
+
+    def _read_config(self, config):
+        if isinstance(config, dict):
+            pass
+        elif isinstance(config, str):
+            config = self._parse(config)
+        else:
+            raise ValueError("config must be of type dict or str, given" +
+                             f"{type(config)}")
+
+        return config
+
+    @staticmethod
+    def _parse(filename):
+        """
+        Parse a configuration file.
+
+        Parameters
+        ----------
+        filename : str
+            The config file to parse. Should be YAML formatted.
+
+        Returns
+        -------
+        config: dict
+            The raw config file as a dictionary.
+        """
+
+        with open(filename, 'r') as fp:
+            config_str = jinja2.Template(fp.read()).render()
+        config = yaml.load(config_str, Loader=yaml.Loader)
+
+        return config
+
+    def create_sacc_cov(self, cov, output='cls_cov.fits'):
+        """
+        Write created cov to a new sacc object
+
+        Parameters:
+        ----------
+        output (str): filename output.
+
+        Returns:
+        -------
+        None
+
+        """
+        output = os.path.join(self.get_outdir(), output)
+
+        s = self.get_sacc_file().copy()
+        s.add_covariance(cov)
+        s.save_fits(output, overwrite=True)
+
+        return s
+
+    def get_outdir(self):
+        return self.outdir
+
+    def get_sacc_file(self):
+        if self.sacc_file is None:
+            sacc_file = self.config['tjpcov'].get('sacc_file')
+            if isinstance(sacc_file, sacc.Sacc):
+                self.sacc_file = sacc_file
+            elif isinstance(sacc_file, str):
+                self.sacc_file = sacc.Sacc.load_fits(sacc_file)
+            else:
+                raise ValueError("sacc_file must be a sacc.Sacc or str, " +
+                                 f"given {type(sacc_file)}")
+
+        return self.sacc_file
+
+    def get_sacc_with_concise_dtypes(self):
+        """
+        Return a copy of the sacc file with concise data types
+
+        Parameters:
+
+        Returns:
+        --------
+            sacc_data (Sacc): Data Sacc instance with concise data types
+        """
+
+        s = self.get_sacc_file().copy()
+        dtypes = s.get_data_types()
+
+        dt_long = []
+        for dt in dtypes:
+            if len(dt.split('_')) > 2:
+                dt_long.append(dt)
+
+        for dt in dt_long:
+            pd = sacc.parse_data_type_name(dt)
+
+            if pd.statistic != 'cl':
```

Comment:
If seems this will not work for real space (xi) or cluster counts to be implemented

Reply:
Moved to the Fourier base class